### PR TITLE
Fix subject number in aas_addcontrast

### PR DESCRIPTION
**Pull request by TamerGezici**
I'm not 100% sure if this is a correct fix (or if it is optimal, as I am not a huge fan of MATLAB so my code may not  be the best MATLAB implementation) and/or if it may fail for some situations.

But this code works fine for me. Can you please review it?

I mentioned the issue here before: https://github.com/automaticanalysis/automaticanalysis/issues/336#issuecomment-1160843159

And I can elaborate if my explanation there is unclear.

### CODE REVIEW

**Comment by tiborauer** [comment]
I have looked at your changes, and I think it will not work if subjects have different sets of series because `subj_id` is set outside/before the loop.

**Comment by TamerGezici** [comment]
> I have looked at your changes, and I think it will not work if subjects have different sets of series because `subj_id` is set outside/before the loop.

I see. True. I forgot that subjname is a list of subjects. Let me see what I can do. 

However,  there is an issue in this code: particularly when you call addsubject one by one for subject at a time in a loop.

**Comment by tiborauer** [comment]
I am not sure how adding subjects one at a time in a loop is a problem.

**Comment by TamerGezici** [comment]
I will try explaining with screenshots later.

Assume we are looping over our 3rd subject, using aas_addcontrast.

The function receives an array of subject names. The loop (https://github.com/automaticanalysis/automaticanalysis/blob/331702ec9c23307c44012ad2b12b9cf735b36c17/aa_engine/aas_addcontrast.m#L265) loops over each subject in the array. When you pass only one subject, the `numel `is 1.

The loop initializes with `subj `value 1. And will always stay as that because I have provided only one subject. The variable subj is passed onto `aas_getseries `(https://github.com/automaticanalysis/automaticanalysis/blob/331702ec9c23307c44012ad2b12b9cf735b36c17/aa_engine/aas_addcontrast.m#L279)

aas_getseries receives `subj` and indexes subject list like this: `aap.acq_details.subjects(subj).seriesnumbers;`

or when using literals instead of variables, like this: `aap.acq_details.subjects(1).seriesnumbers;`

The series numbers of subject 1 is taken instead of the subject's real series number.

**Comment by tiborauer** [comment]
I see. Yes, it is an issue, indeed.

**Comment by tiborauer** [comment]
We could replace `[~, mriser] = aas_get_series(aap,'functional',subj,sess);` with `[~, mriser] = aas_get_series(aap,'functional',strcmp({aap.acq_details.subjects.subjname},subjname{subj}),sess);`

**Comment by TamerGezici** [comment]
> We could replace `[~, mriser] = aas_get_series(aap,'functional',subj,sess);` with `[~, mriser] = aas_get_series(aap,'functional',find(strcmp({aap.acq_details.subjects.subjname},subjname{subj})),sess);`

Aha, indeed. Let me try this.

**Comment by TamerGezici** [comment]
I haven't tested this yet. I committed to test for myself on my other computer. I will tag you when I replicate previous analysis.

File: aa_engine/aas_addcontrast.m [hunk 1]
```diff
@@ -262,6 +262,14 @@
     subjname = {aap.acq_details.subjects.subjname};
 end
 
+sub_id = 0;
```
Comment (tiborauer) [inline aa_engine/aas_addcontrast.m:265]: I guess we do not need this loop any more.

Comment (TamerGezici) [reply aa_engine/aas_addcontrast.m:265]: Indeed. I'll remove it on the next commit.

**Comment by TamerGezici** [comment]
The code is fine for me. I was able to replicate the same analysis I've done before with that loop I'd implemented on my local branch.

**Comment by tiborauer** [comment]
Please merge the latest changes for your checks to be successful.

**Comment by TamerGezici** [comment]
> Please merge the latest changes for your checks to be successful.

Done. Though we only changed one line, since I have been sending commits in between, the fork registers as `This branch is [4 commits ahead]`

If this is a big problem (uselessly increasing the commit count whereas it's just 1 new line added) I guess I can re-base the branch and re-commit.

**Pull request merged**